velox/dwio/parquet/thrift/ParquetThriftTypes.h

```diff
@@ -318,7 +318,8 @@ struct CompressionCodec {
     BROTLI = 4,
     LZ4 = 5,
     ZSTD = 6,
-    LZ4_RAW = 7
+    LZ4_RAW = 7,
+    QPL = 8,
```

Comment:
This file is automatically generated from the `parquet.thrift` file. Adding extra codec here breaks the parquet specs. BTW, is `QPL` a codec type or implementation of one or more codec types?

Reply:
@wgtmac The QPL supports de/compression compatible with the Deflate compression standard described in **RFC1951**, it can de/compress each other with zlib library if set the history buffer <= 4K and set deflate mode(not gzip or zlib mode). As for the issue you proposed, I will spend some time to find a solution to avoid adding a codec type in **parquet.thrift** file.

Reply:
Parquet supports gzip, the gzip is based on deflate, the difference of gzip and deflate is window_bits which is put in **deflateInit2** function, just like decompression.

- window_bits < 0-----> deflate
- 0<window_bits<=15-----> zlib 
- window_bits>=16-----> gzip

`      constexpr int WINDOW_BITS = 15;
      // Determine if this is libz or gzip from header.
      constexpr int DETECT_CODEC = 32;
      // Initialize decompressor.
      auto ret = inflateInit2(&stream, WINDOW_BITS | DETECT_CODEC);`

Reply:
Maybe not, because default gzip history buffer is 32KB, but qpl only supports history buffer <=4KB,  qpl can not decompress the file compressed by standard gzip if default history buffer. I will report to my leader and hope for some good solutions.

Reply:
@wgtmac  @Yuhta  , QPL is not fully compatible with zlib now due to hardware limitation, maybe it will be solved in the future.  So looks like if we want to enable IAA for Velox, the prerequisite is we need request to add a new codec in parquet format. Am I right?  As an alternative, we can use another Intel accelerator [QAT](https://www.intel.com/content/www/us/en/architecture-and-technology/intel-quick-assist-technology-overview.html). It is fully compatible with gzip codec defined in parquet format [GZIP](https://github.com/apache/parquet-format/blob/c766945d90935ebcd4e03fee13aad2b6efcadce3/Compression.md?plain=1#L54). So it can acclerate velox parquet read without a new codec. Do you think it's a good idea to enable Velox parquet reader with QAT.